src/momento/aio/_add_header_client_interceptor.py

```diff
@@ -32,13 +32,51 @@ async def intercept_unary_unary(
         client_call_details: grpc.aio._interceptor.ClientCallDetails,
         request: grpc.aio._typing.RequestType,
     ) -> Union[grpc.aio._call.UnaryUnaryCall, grpc.aio._typing.ResponseType]:
+
+        # FIXME Hack to account for ddtrace lib injecting grpc metadata as list instead of proper grpc.aio.Metadata()
+        # Will open issue on data dog repo to discuss this more.
+        # Since once client_call_details.metadata is set as list we cant overwrite with `Metadata()` class we build
+        # up a new ClientCallDetails() in each request now in this interceptor and make sure Metadata() is set properly
```

Comment:
It might be worth copying a tiny bit of the context from your commit message into this comment.  At least the part about your proposed fixes, which I largely agree with.

Or maybe better to create a github issue that has your proposed fixes and then link that issue in the comment here.

Reply:
will make issue and link